addons/mrp/models/stock_move.py

```diff
@@ -293,6 +293,24 @@ def _consuming_picking_types(self):
         res.append('mrp_operation')
         return res
 
+    def _get_qty_to_split(self):
+        qty_to_split = super()._get_qty_to_split()
+        # In case of backorder after overconsumption of components, we don't want to take the remaining quantities of every raw moves
+        # but the quantities needed to produce what's left to be produced
+        production = self.env['mrp.production'].search([('id', '=', self.env.context.get('bo_production_id'))])
```

Comment:
Why you use the context here? (`raw_material_production_id` | `production_id`) should contain the same information ? 

(technically) Also, avoid using a search to get a record from id -> Use browse instead:
production = self.env['mrp.production'].browse(int(self.env.context.get('bo_production_id')))
(avoid a useless SQL query + cleaner)